src/resdk/tables/qc.py

```diff
@@ -40,18 +41,27 @@
 
 
 def _filter_and_rename_columns(df, column_map):
-    """Filter and rename columns based on provided specifications."""
-    valid_columns = [
-        col.get("name", "") for col in column_map if col.get("name", "") in df.columns
-    ]
-    df = df[valid_columns]
-
-    rename_map = {
-        col["name"]: col["slug"]
-        for col in column_map
-        if col.get("name", "") in df.columns
-    }
-    df = df.rename(columns=rename_map)
+    """Filter, apply a scaling factor and rename columns based on provided specifications."""
+    selected_columns = []
+    rename_map = {}
+
+    for col in column_map:
+        names = col.get("name", [])
+        slug = col.get("slug", "")
+        scaling_factor = col.get("scaling_factor", {})
+
+        if isinstance(names, str):
+            names = [names]
+
+        for name in names:
+            if name in df.columns:
+                selected_columns.append(name)
+                rename_map[name] = slug
+                # Some column values are given a specific format (e.g. millions)
+                # and need to be scaled on a common scale
+                df[name] = df[name] * scaling_factor.get(name, 1)
```

Comment:
Once a name is selected further search over names makes no sense. Add `break` to exit the inner for loop.